homeassistant/components/simplisafe/binary_sensor.py

```diff
@@ -51,11 +51,12 @@ async def async_setup_entry(hass, entry, async_add_entities):
 
     sensors = []
     for system in simplisafe.systems.values():
-        for sensor in system.sensors.values():
-            if sensor.type in SUPPORTED_SENSOR_TYPES:
-                sensors.append(SimpliSafeBinarySensor(simplisafe, system, sensor))
-            if sensor.type in SUPPORTED_BATTERY_SENSOR_TYPES:
-                sensors.append(SimpliSafeSensorBattery(simplisafe, system, sensor))
+        if system.version == 3:
```

Comment:
Why are we checking the system version every time through the loop? I would check for system version before this loop and if version 2 is found, log a message (`INFO`-level) and return without doing anything else.

Reply:
I'm doing it in the loop because I assume a customer may have a SimpliSafe 2 system *and* a SimpliSafe 3 system in one account, meaning that one would work and the other wouldn't. Do you agree @bachya ?

Reply:
Apologies, my language was unclear – we still need a loop, but we can short-circuit it sooner for V2 systems:

```python
async def async_setup_entry(hass, entry, async_add_entities):
"""Set up SimpliSafe binary sensors based on a config entry."""
    simplisafe = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
    sensors = []

    for system in simplisafe.systems.values():
        if system.version == 2:
            _LOGGER.info("Skipping sensor setup for V2 system: %s", system.system_id)
            continue

        for sensor in system.sensors.values():
            if sensor.type in SUPPORTED_SENSOR_TYPES:
                sensors.append(SimpliSafeBinarySensor(simplisafe, system, sensor))
            if sensor.type in SUPPORTED_BATTERY_SENSOR_TYPES:
                sensors.append(SimpliSafeSensorBattery(simplisafe, system, sensor))

    async_add_entities(sensors)
```